modules/core/src/lib/attribute.js

```diff
@@ -200,15 +192,15 @@ export default class Attribute extends BaseAttribute {
       if (this.buffer && this.buffer.byteLength < this.value.byteLength) {
         this.buffer.reallocate(this.value.byteLength);
 
-        if (shouldCopy) {
+        if (shouldCopy && oldValue) {
           // Upload the full existing attribute value to the GPU, so that updateBuffer
           // can choose to only update a partial range.
           // TODO - copy old buffer to new buffer on the GPU
           this.buffer.subData(oldValue);
         }
       }
 
-      this.setNeedsUpdate(true, {startRow: instanceCount});
```

Comment:
Why redraw is not needed?

Reply:
Redraw and update are two different things. `allocate` is only called during update. The line was excessive.